redis/pool.go

```diff
@@ -376,6 +359,52 @@ func (p *Pool) lazyInit() {
 	p.mu.Unlock()
 }
 
+// waitVacantConn waits until at least one vacant connection gets spotted in the pool.
```

Comment:
```suggestion
// waitVacantConn waits for a vacant connection in pool if waiting
// is enabled and pool size is limited, otherwise returns instantly.
```